### ISSUE TITLE
Live View fine when away from home but not when home

### ISSUE THREAD

**Issue by Crox77** [dgreif/ring#352]
# Live Streaming Issue

### Describe the live streaming issue
<Bit of an odd one, but everything has worked fine up until the last couple of days. I am unable to view my live stream when im at home and connected to my wifi. it does however work fine when I am away from home. 
Running the latest update and all notifications work fine> A clear and concise description of the problem.  If all you can say is that "live streaming doesn't work on my network", then please do not open an issue.

One thing to add when im home and try to connect Homebridge logs say live stream has started but it times out in the Home app and says no response?

I have also made sure live stream works in the Ring app all the time (it does) and all modes that could upset streaming is turned off.

It also behaves the same on my iPhone and iAd and the wifes iPhone .

would it be worth me reseting the doorbell and setting up again?

Any Ideas please?
Thanks

### Screenshots/Logs
 If applicable, add screenshots and log output to help explain your problem.

### Recommended solution
 Please describe what you think is causing this issue, and how you recommend it be fixed

### Environment

 - Ring Camera: [Ring Doorbell 1]
 - OS: [Mac OS]
 - Node.js: [v12.16.3] 
 - NPM: [v6.14.4] 
 - homebridge-ring (if applicable): [v9.0.1]
 - homebridge (if applicable): [homebridge-config-ui-x v4.21.0]
 - hoobs (if applicable): [e.g. 0.4.50 or N/A]

### Network Environment
 Please provide a detailed description of your network setup, including all network hardware between your camera, the internet, your homebridge server, and your HomeKit home hub (Apple TV, HomePod, iPad).  Please include brand and model for each device that the network traffic passes through.

**Comment by Crox77** [comment]
any ideas @dgreif?

**Comment by dgreif** [comment]
@Crox77 this is an odd one because I have several other users in Discord complaining that it works when they are on WiFi, but not when they are on LTE, which is the exact opposite of your issue.  In your case, I'd make sure that your phone is on the same subnet as your homebridge server.  If you want a technical guess, I think it's a matter of the video stream sending the first keyframe before your phone is ready to receive it, and then it takes too long before the next keyframe which cause HomeKit to time out.  I'll probably mess around with the keyframe timing a bit in the future, but I don't think there is anything I can do for your particular case at the moment.

**Comment by Crox77** [comment]
@dgreif Thanks for your reply, Very odd one I agree, it has been fine until the weekend, one thing id add though is if I switch my Wifi off at home and connect on 4g it still does not stream only streams when im away from my home

**Comment by dgreif** [comment]
Yeah, it's probably all network related then.  Probably nothing I can do from the plugin side

**Comment by Crox77** [comment]
is it worth me removing the doorbell from ring app and setting up again?

**Comment by Crox77** [comment]
@dgreif  oddly ive just started getting these errors in the log as we have been talking.

[6/9/2020, 7:01:47 PM] [Ring] Snapshot failed to refresh after 40 attempts
[6/9/2020, 7:01:48 PM] [Ring] 404 from endpoint https://api.ring.com/clients_api/snapshots/image/22037958
[6/9/2020, 7:01:48 PM] [Ring] Failed to cache snapshot for Front Door (26.246s), The camera currently reports that is it online
[6/9/2020, 7:01:48 PM] [Ring] Front Door camera appears to be unable to upload snapshots.  This can happen when your Ring Modes settings prevent motion detection or live view, which also prevents snapshots.  If Modes are not blocking snapshots, this can also happen if the camera has gotten into a bad state, which requires a physical restart of the camera.  If this happens, you will also not be able to retrieve snapshots via the Ring app for this camera.  In that case, please turn off power to this camera by removing its battery or turning off the breaker for the circuit it is wired to.  Once power is cycled, snapshots should start working again.

Im a bit unsure how to power cycle, does it mean reseting with the orange button? and do I then need to add to Ring app again?

**Comment by Crox77** [comment]
@dgreif  one thing I’ve noticed this morning when viewing my door bell away from home. The Mic icon has disappeared from the viewing screen.
I’m thinking about uninstalling everything and adding to ring and HomeKit again.
Do you have any thoughts?

**Comment by JazzerAlto** [comment]
Just throwing in an observation.  Can you try it again this morning?  As Dusty had mentioned, I was having a terrible time over LTE but this morning, my live streams are working great over LTE so it might have been something wonky with the Ring Servers and/or some internet routing issues.  I didn't make any changes, it just started working so it might be worth to check you situation again for improvement.

**Comment by Crox77** [comment]
So bit of an update, I wasn’t prepared to take my ring doorbell apart to power cycle it Nor did I like the idea of removing from the ring app and setting up again.
I had tried rebooting my HomePod but still I couldn’t get to view live camera when home and was resounded to just being able to view while away from home.
Anyhow I came home today and my Network was so sluggish I decided I’d switch my router, iMac and HomePod all off.
Restarted things one at a time then restarted Homebridge. 
Checked the Doorbell and it came up no response, force closed the home app, tried it again and it worked strait away! And it continues to be ok as we speak.
I will keep an eye on it and close this thread.
Thank you @dgreif for your help and continued work on this plugin
